metrics/batch.go

```diff
@@ -27,10 +30,27 @@ func NewBatch(source string) *Batch {
 func (b *Batch) AddGauge(g librato.Gauge) {
 	b.mutex.Lock()
 	b.Librato.Gauges = append(b.Librato.Gauges, g)
+	b.modified = true
+	b.modifiedAt = time.Now()
 	b.mutex.Unlock()
 }
+
 func (b *Batch) AddCounter(g librato.Counter) {
 	b.mutex.Lock()
 	b.Librato.Counters = append(b.Librato.Counters, g)
+	b.modified = true
+	b.modifiedAt = time.Now()
+	b.mutex.Unlock()
+}
+
+func (b *Batch) Flush(d time.Duration) *librato.Batch {
+	var lb *librato.Batch
+	b.mutex.Lock()
+	if b.modified && time.Since(b.modifiedAt) > d {
```

Comment:
Couldn't this strategy lead to duplicate gauge/counter in the batch if the intervals are incorrect?